roles/heat/tasks/main.yml

```diff
@@ -70,6 +44,15 @@
     - heat-api-cfn
     - heat-engine
 
+- name: heat config
+  template: src=etc/heat/{{ item }} dest=/etc/heat/{{ item }} mode=0644
+  with_items:
+    - heat.conf
+    - policy.json
+    - api-paste.ini
+  notify:
+    - restart heat services
```

Comment:
Heat configs need to be put in place before a database sync is attempted. You've changed the order here so the db sync will fail.